internal/daemon/plugin/vendorplugin.go

```diff
@@ -131,8 +153,10 @@ func NewGrpcPlugin(dpuMode bool, client client.Client, opts ...func(*GrpcPlugin)
 		opt(gp)
 	}
 
-	if gp.vspImage == "" {
+	if gp.vsp.VendorSpecificPluginImage == "" {
 		gp.log.Info("WARNING: VSP Image not set, skipping vendor plugin container startup")
+	} else {
+		gp.deployVspDs()
```

Comment:
I think the `if` check should move inside `deployVspDs()`.

Reply:
Yeah good point, updated